src/hotspot/share/gc/shared/cardTable.cpp

```diff
@@ -208,6 +208,18 @@ void CardTable::dirty_MemRegion(MemRegion mr) {
 }
 
 void CardTable::clear_MemRegion(MemRegion mr) {
+  // The MemRegion mr can have a word size of 0. This occurs the first time
+  // a SerialGC full collection is performed, for example. In that case, the
```

Comment:
Why does the first full collection result in a MemRegion of word size 0?

Reply:
> Why does the first full collection result in a MemRegion of word size 0?

[CardTableRS::maintain_old_to_young_invariant](https://github.com/openjdk/jdk/blob/c856b3425a70d2aecb6c5c44da36396a5d74b00d/src/hotspot/share/gc/serial/cardTableRS.cpp#L51) calls `clear_MemRegion(old_gen->prev_used_region());` and the `prev_used_region()` is an empty region the first time this method is called for a full collection.